src/mockServiceWorker.js

```diff
@@ -254,11 +257,12 @@ async function getResponse(event, client, requestId) {
       redirect: request.redirect,
       referrer: request.referrer,
       referrerPolicy: request.referrerPolicy,
-      body: await request.text(),
+      body,
       bodyUsed: request.bodyUsed,
       keepalive: request.keepalive,
     },
-  })
+  }
+  const clientMessage = await sendToClient(client, message, body)
```

Comment:
But the `body` property is not set conditionally on `payload`. So it's always there. Then, we can check its value as `typeof payload.body !== 'undefined'` (or `== null` since we're coercing it to null if it's empty) in the `sendToClient` to achieve the same result, can't we?

Reply:
We can. Let me try that

Reply:
Go ahead.

It's be great not to duplicate things, as two arguments imply two different values being passed. Since `body` is the same as `message.body` and really shouldn't differ intention-wise, I think an extra argument creates a confusion.

Reply:
Done